reconcile/dashdotdb_dvo.py

```diff
@@ -149,18 +187,64 @@ def _get_clusters(cnfilter=None):
             return [result for result in results if result['name'] == cnfilter]
         return results
 
+    def _get_token(self):
+        params = {'scope': 'deploymentvalidation'}
+        endpoint = (f'{self.dashdotdb_url}/api/v1/'
+                    f'token')
+        response = requests.get(url=endpoint,
+                                params=params,
+                                auth=(self.dashdotdb_user,
+                                      self.dashdotdb_pass),
+                                timeout=(5, 120))
+        try:
+            response.raise_for_status()
+        except requests.exceptions.RequestException as details:
+            LOG.error('%s error retrieving token for DVO data: %s',
+                      self.logmarker, details)
+            return None
+        return response.text.replace('"', '').strip()
+
+    def _close_token(self, token):
+        params = {'scope': 'deploymentvalidation'}
+        endpoint = (f'{self.dashdotdb_url}/api/v1/'
+                    f'token/{token}')
+        response = requests.delete(url=endpoint,
+                                   params=params,
+                                   auth=(self.dashdotdb_user,
+                                         self.dashdotdb_pass),
+                                   timeout=(5, 120))
+        try:
+            response.raise_for_status()
+        except requests.exceptions.RequestException as details:
+            LOG.error('%s error closing token for DVO data: %s',
+                      self.logmarker, details)
+
     def run(self, cname=None):
-        validation_list = ('operator_replica', 'operator_request_limit')
-        for validation in validation_list:
-            LOG.debug('%s Processing validation: %s',
-                      self.logmarker, validation)
-            validations = threaded.run(func=self._get_deploymentvalidation,
+        validation_list = threaded.run(func=self._get_validation_names,
                                        iterable=self._get_clusters(cname),
                                        thread_pool_size=self.thread_pool_size,
-                                       validation=validation)
-            threaded.run(func=self._post,
-                         iterable=validations,
-                         thread_pool_size=self.thread_pool_size)
+                                       filter='deployment_validation_operator')
+        if validation_list:
+            validation_names = {v['cluster']: v['data']
+                                for v in validation_list if v}
+        clusters = self._get_clusters(cname)
+        token = self._get_token()
```

Comment:
I thought about doing this, but I think I didn't do it because I felt it was more variable than the other things.  I've no issue making it a class var though.

Reply:
Done